076-connector-offsets-support.md

```diff
@@ -0,0 +1,286 @@
+
+```yaml
+listOffsets:
+  toConfigMap:
+    name: my-connector-offsets
+```
+
+The ConfigMap will contain a single entry called `offsets.json` that contains the response received from the GET /connectors/{connector}/offsets endpoint.
+
+For example for a source connector the ConfigMap might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-source-connector-offsets
+  ownerReferences:
+    - apiVersion: kafka.strimzi.io/v1beta2
+      blockOwnerDeletion: false
+      controller: false
+      kind: KafkaConnector
+      name: my-source-connector
+      uid: 1234
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            // Connector-defined source partition
+          },
+          "offset": {
+             // Connector-defined source offset
+          }
+        }
+      ]
+    }
+```
+
+For a sink connector it might look like:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-sink-connector-offsets
+  ownerReferences:
+    - apiVersion: kafka.strimzi.io/v1beta2
+      blockOwnerDeletion: false
+      controller: false
+      kind: KafkaConnector
+      name: my-sink-connector
+      uid: 5678
+data:
+  offsets.json: |
+    {
+      "offsets": [
+        {
+          "partition": {
+            "kafka_topic": "my-topic"
+              "kafka_partition": 2
+          },
+          "offset": {
+            "kafka_offset": 4
+          }
+        }
+      ]
+    }
+```
+
+Notes:
+* If the `listOffsets` property is missing from the KafkaConnector CR when the `strimzi.io/connector-offsets=list` annotation is added the operator will update the KafkaConnector CR status to include a warning message:
+  ```yaml
+  - lastTransitionTime: "2024-06-04T08:44:15.913138115Z"
+    message: Failed to list the connector offsets due to missing property listOffsets in KafkaConnector CR.
+    reason: ListOffsets
+    status: "True"
+    type: Warning
+  ```
+  The operator will leave the annotation on the KafkaConnector resource until either the list operation succeeds, or the user removes the annotation.
+  This means the operator will retry the list on every reconciliation, allowing the condition to remain present for the user to see.
+
+### Altering offsets
+
+To alter offsets the user will annotate the KafkaConnector resource with the new annotation `strimzi.io/connector-offsets` set to `alter`.
+After the annotation is added, on the next reconciliation the operator will call the PATCH /connectors/{connector}/offsets endpoint to alter the offsets.
+Once the patch is complete the operator will then remove the annotation from the KafkaConnector CR.
+
+The operator will read in the entry called `offsets.json` from the ConfigMap.
+The entry name matches the entry name used in the list operation, so that a user can list offsets, then edit the ConfigMap the operator created, then request the offsets be altered using that same ConfigMap.
```

Comment:
```suggestion
The entry name matches the entry name used in the list operation, so that a user can list offsets, then edit the ConfigMap the operator created, then request the offsets to be altered using that same ConfigMap.
```